src/main/java/cceclipseplugin/core/EclipseRequestManager.java

```diff
@@ -0,0 +1,203 @@
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.NullProgressMonitor;
+import org.eclipse.core.runtime.Path;
+import org.eclipse.swt.widgets.Display;
+
+import cceclipseplugin.ui.UIRequestErrorHandler;
+import cceclipseplugin.ui.dialogs.DialogStrings;
+import cceclipseplugin.ui.dialogs.MessageDialog;
+import dataMgmt.DataManager;
+import dataMgmt.models.FileMetadata;
+import dataMgmt.models.ProjectMetadata;
+import requestMgmt.IInvalidResponseHandler;
+import requestMgmt.RequestManager;
+import websocket.IRequestSendErrorHandler;
+import websocket.WSManager;
+import websocket.models.File;
+import websocket.models.Project;
+import websocket.models.Request;
+import websocket.models.requests.FileCreateRequest;
+import websocket.models.requests.FilePullRequest;
+import websocket.models.responses.FileCreateResponse;
+import websocket.models.responses.FilePullResponse;
+
+public class EclipseRequestManager extends RequestManager {
+
+	public EclipseRequestManager(DataManager dataManager, WSManager wsManager,
+			IRequestSendErrorHandler requestSendErrorHandler, IInvalidResponseHandler invalidResponseHandler) {
+		super(dataManager, wsManager, requestSendErrorHandler, invalidResponseHandler);
+	}
+	
+	@Override
+	public void finishSubscribeToProject(long id, File[] files) {
+		IWorkspaceRoot root = ResourcesPlugin.getWorkspace().getRoot();
+		Project p = PluginManager.getInstance().getDataManager().getSessionStorage().getProjectById(id);
+		IProject eclipseProject = root.getProject(p.getName());
+		ProjectMetadata meta =  new ProjectMetadata();
+		meta.setName(p.getName());
+		meta.setProjectID(id);
+		PluginManager.getInstance().getMetadataManager().putProjectMetadata(eclipseProject.getFullPath().toString(), meta);
+		try {
+			NullProgressMonitor progressMonitor = new NullProgressMonitor();
+			if (eclipseProject.exists()) {
+				eclipseProject.delete(true, true, progressMonitor);
+			}
+			eclipseProject.create(progressMonitor);
+			eclipseProject.open(progressMonitor);
+			for (File f : files) {
+				pullFileAndCreate(eclipseProject, p, f, progressMonitor);
+			}
+		} catch (CoreException e) {
+			e.printStackTrace();
+		}
+	}
+	
+	private byte[] fileBytes;
+	
+	public void pullFileAndCreate(IProject p, Project ccp, File file, IProgressMonitor progressMonitor) {
+		Request req = (new FilePullRequest(file.getFileID())).getRequest(response -> {
+				if (response.getStatus() == 200) {
+					fileBytes = ((FilePullResponse) response.getData()).getFileBytes();
+					
+					String path = file.getRelativePath();
+					Path relPath = new Path(path);
+					if (!path.equals("") && !path.equals(".")) {
+						String currentFolder = "";
+						for (int i = 0; i < relPath.segmentCount(); i++) {
+							// iterate through path segments and create if they don't exist
+							currentFolder = Paths.get(currentFolder, relPath.segment(i)).toString();
```

Comment:
Any reason we don't maintain this as a Path variable and then call .toString() on it when it needs to be a string?

Storing string-based paths is discouraged where there is a class that is designed to represent file-system paths.

So something like Paths.resolve should allow us to append a string to a path, and get a new path. Saves the conversion to and from a string every single iteration.